app/javascript/lib/validation/validate-file-content.js

```diff
@@ -235,29 +315,35 @@ function sniffDelimiter([line1, line2], mimeType) {
 }
 
 /**
- * Verify cap format for a cluster or metadata file
+ * Verify cap format for a cluster, metadata, or expression matrix file
  *
- * The "cap" of an SCP study file is its first two lines, i.e.:
+ * The "cap" of an SCP study file is its first "few" lines that contain structural data., i.e.:
  *  - Header (row 1), and
  *  - Annotation types (row 2)
  *
  * Cap lines are like meta-information lines in other file formats
  * (e.g. VCF), but do not begin with pound signs (#).
  */
```

Comment:
BLOCKER: we're trying to avoid branches based on file type in individual validation methods so that they are easier to reason about.  So I would leave this method as-is, and just create a new `validateDenseHeader` method that checks that the header line exists and then calls `validateGeneInHeader`